tests/integration/device_test.sh

```diff
@@ -16,67 +16,39 @@ function device_info_test_helper()
   local fs_size
   local output
   local expected_output
+  local container_tmp_dir
+  local kw_device_simulation="${CONTAINER_DIR}/scripts/kw_device_expected_output.sh"
 
-  # buffer with all information.
-  container="kw-${distro}"
-  buffer=$(container_exec "${container}" 'kw device --local')
+  for distro in "${DISTROS[@]}"; do
+    container="kw-${distro}"
 
-  # some fields must be ignored because they surely won't match.
-  filter=(
-    'Root filesystem' 'Size' 'Mounted on'                                            # storage fields
-    'Distribution' 'Distribution base' 'Distribution version' 'Desktop environments' # desktop fields
-  )
+    container_tmp_dir=$(container_exec "$container" 'mktemp --directory')
 
-  # add | separator to each filter item.
-  filter_regex=$(sed 's/ /|/g' <<< "${filter[@]}")
+    # Copy the kw_device_expected.sh to the container's workdir directory
+    container_copy "$container" "$kw_device_simulation" "$container_tmp_dir"
 
-  # the actual perl regex.
-  filter_regex="(${filter_regex}):"
+    # Get the output of the `kw device --local` command from the container
+    buffer=$(container_exec "$container" 'kw device --local')
 
-  # get the output, after applying filter.
-  output=$(printf '%s' "${buffer}" | grep --invert-match --perl-regexp "${filter_regex}")
+    # Specify the fields to filter out from the output
```

Comment:
Consider adding a sample example of the type of output that this filter will work on.